workshops/payments/views.py

```diff
@@ -303,13 +305,31 @@ def post(self, request, *args, **kwargs):
                 body += '<li>%s (%s): %s</li>' % (oi.name, oi.email, oi.rate)
             body += '</ul></div>'
             body += '</html>'
-
+            rates = {}
+            for ticket in all_orders:
+                rate = ticket.rate
+                if rate not in rates:
```

Comment:
When does this case happen?

Reply:
No idea, it comes from the same structure on line 205.

Reply:
I didn't write that, it came from the same structure on line 205, but it looks like that's supposed to determine what number of each type of order they purchased correct? So when that order type is first seen, it initializes a dictionary entry counting the number of instances of that order to one, then increments it for every subsequent encounter of that order type doesn't it?

Reply:
Yep, I recognize the code, but, I don't think we need any of that aggregation here, right? Each ticket should just be a line-item in the invoice, no need to summarize the data.

Reply:
So if they buy 3 of one kind of ticket and 2 of another, we want five lines, one for each ticket, instead of one line for the first type and another line for the other type?

Reply:
Yeah, let's do it that way (and include the ticket-holder's info in each line item).